Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+    ax=axes('Position', canvas, 'Visible', settings.axesVisible, 'Parent', figure);
+    [img, ~, alphachannel] = imread(ImagePath);
+    fg= imshow(img);
+
+    if strcmp('png', ImagePath(end-2:end))
+        fg.AlphaData=alphachannel;
+    end
+
+    clear fg
+    settings.figurecount = xASL_sub_PrintHeader(header, figure, settings, canvas);
+end
+
+function [strout] = xASL_sub_replace(strin, x)
+    strout = strin;
+    substring = regexp(strin, '<\w*>', 'match');
+    for substringIndex=1:length(substring)
+        if ~isfield(x, substring{substringIndex}(2:end-1))
+            warning(['Could not replace ', substring{substringIndex}, ' check if file exists in ExploreASL/Derivatives/Population']);
+        else
+            strout = strrep(strout, substring{substringIndex}, x.(substring{substringIndex}(2:end-1)));
+        end
+    end
+end
+
+function [settings] = xASL_sub_PrintScan(input, x, figure, settings)
+    position = [xASL_str2num(input.position) xASL_str2num(input.size)];
+    [canvas, ~] = xASL_sub_createNewCanvas(position(1:2), position(3:4), settings.canvas);
+
+    if ~isfield(x.P, input.name)
+        warning (['could not print', input.name, 'check if nifti exists in ExploreASL/Derivatives/Population']);
+        return
+    end
+
+    ImIn = {x.P.(input.name)};
+    header = input.name;
+    ax=axes('Position', canvas, 'Visible', settings.axesVisible, 'Parent', figure);
+    
+    if isfield(input, 'overlay')
+        fields = fieldnames(input.overlay);
+        for iField=1:length(fields)
+            if isfield(x.P, fields(iField))
+                ImIn(end+1) = {x.P.(fields{iField})};
+                header = [header ' + ' fields{iField}];
+            end
+        end
+    end
+
+    if isfield(input, 'slice')
+        [x.S.TraSlices] = xASL_sub_getSliceFromStruct(input.slice, 'TraSlice');
+        [x.S.CorSlices] = xASL_sub_getSliceFromStruct(input.slice, 'CorSlice');
+        [x.S.SagSlices] = xASL_sub_getSliceFromStruct(input.slice, 'SagSlice');       
+    else
+        x.S.TraSlices = [25, 50, 90];
+        x.S.SagSlices = [25, 50, 90];
+        x.S.CorSlices = [25, 50, 90];
+    end
+
+    img = xASL_vis_CreateVisualFig(x, ImIn, [], [], [], [], [], [], [], [], [], [], []);
+    fg = imshow(img);
+    clear fg
+    settings.figurecount = xASL_sub_PrintHeader(header, figure, settings, canvas);
+end
+
+function [slice] = xASL_sub_getSliceFromStruct(struct, name)
+
+    if isfield(struct, name) && ~isempty(struct.(name))
+        slice = str2num(struct.(name));
+    else 
+        slice = [];
+    end
+end
+
+function [newCanvas, line] = xASL_sub_createNewCanvas(position, size, oldCanvas)
+    newPosition = [oldCanvas(1) + position(1) * oldCanvas(3) oldCanvas(2) + position(2) * oldCanvas(4)];
+    newSize = [oldCanvas(3:4) .* size];
+    newCanvas = [newPosition, newSize];
+    line = [newPosition(1)  newPosition(2) + newSize(2) newSize(1) 0];
+end
+
+function line = xASL_sub_PrintText(input, figure, line, settings)
```

Comment:
above you have clearly defined types, and now you have an undefined first argument "input", which is handled differently depending on its content, I would avoid this, this makes for difficult to read code

Reply:
This is mostly to avoid duplication, as of right now the watermark (This is generated by ExploreASL on the bottom of the page) is handled by the same function as the function that parses the pdf report. We can discuss this later.